core/modules/memman/MemMan.h

```diff
@@ -0,0 +1,177 @@
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_QSERV_MEMMAN_MEMMAN_H
+#define LSST_QSERV_MEMMAN_MEMMAN_H
+
+// System headers
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace lsst {
+namespace qserv {
+namespace memman {
+
+//-----------------------------------------------------------------------------
+//! @brief Describe a table that can be potentially locked in memory.
+//-----------------------------------------------------------------------------
+
+class TableInfo {
+public:
+
+    std::string tableName;    //< Name of the table
+
+    enum LockOptions {
+        NOLOCK,               //< Item should not be locked
+        MUSTLOCK,             //< Item must be locked or declare failure
+        FLEXIBLE              //< Item may be locked if possible
+    };
+
+    LockOptions theData;      //< Lock options for the table's data
+    LockOptions theIndex;     //< Lock options for the table's index, if any
+
+    //-----------------------------------------------------------------------------
+    //! Constructor
+    //!
+    //! @param  tabName   is the name of the table.
+    //! @param  optData   lock options for the table's data
+    //! @param  optIndex  lock options for the table's index
+    //-----------------------------------------------------------------------------
+
+    TableInfo(std::string const &tabName,
+              LockOptions optData=MUSTLOCK,
+              LockOptions optIndex=NOLOCK)
+             : tableName(tabName), theData(optData), theIndex(optIndex)
+             {}
+};
+
+//-----------------------------------------------------------------------------
+//! @brief The memory manager.
+//!
+//! The MemMan is an abstract class the defines the interface to the memory
+//! manager that is used to lock database chunks in memory.
+//-----------------------------------------------------------------------------
+
+class MemMan {
+public:
+
+    //-----------------------------------------------------------------------------
+    //! @brief Create a memory manager and initialize for processing.
+    //!
+    //! @param  maxBytes   - Maximum amount of memory that can be used
+    //! @param  dbPath     - Path to directory where the database resides
+    //!
+    //! @return !0: The pointer to the memory manager.
+    //! @return  0: A manager could not be created.
+    //-----------------------------------------------------------------------------
+
+    static MemMan *create(unsigned long long maxBytes,
```

Comment:
Prefer the new c++11 fixed-width integer types for "large" values, unless you have an argument otherwise?